spec.md

```diff
@@ -1448,6 +1476,62 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 |-----------|-----------|-------------|-------------------|
 | Invalid `starting_token` | 10 ABORTED | Indicates that `starting_token` is not valid. | Caller SHOULD start the `ListVolumes` operation again with an empty `starting_token`. |
 
+#### `GetVolume`
+
+**EXPERIMENTAL FEATURE**
+
+A Controller Plugin MUST implement this `GetVolume` RPC call if it has `GET_VOLUME` capability.
+
+A Controller Plugin SHOULD provide a non-empty `volume_health` field in `GetVolumeResponse` if it has `VOLUME_HEALTH` capability.
+
+`GetVolumeResponse` should contain current information of a volume if it exists.
+If the volume does not exist any more, `GetVolume` should return gRPC error code `NOT_FOUND`.
+
+The `VolumeHealth` message in `VolumeStatus` field in `GetVolumeResponse` can indicate additional health information for the volume.
+
+Whether the volume is still attached to a node is already handled by `LIST_VOLUMES_PUBLISHED_NODES` controller capability.
+This capability MAY be used by CO to reconcile - re-attach the volume if the volume is not attached any more.
+
+```protobuf
+message GetVolumeRequest {
+  option (alpha_message) = true;
+
+  // Identity information for a specific volume. This field is
+  // REQUIRED. GetVolume will return with current volume information.
+  string volume_id = 1;
+}
+
+message GetVolumeResponse {
+  option (alpha_message) = true;
+
+  message VolumeStatus{
+    // volume_health shows error conditions reported by the SP.
+    // This field MUST be specified if the
+    // VOLUME_HEALTH controller capability is supported.
+    VolumeHealth volume_health = 1 [(alpha_field) = true];
+  }
+
+  // This field is REQUIRED
+  Volume volume = 1;
+
+  // This field is OPTIONAL. This field MUST be specified if the
+  // VOLUME_HEALTH controller capability is supported.
+  VolumeStatus status = 2;
+}
+```
+
+##### GetVolume Errors
+
+If the plugin is unable to complete the GetVolume call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+
+##### VolumeHealth from Controller vs Node Plugins
+
+A SP can implement `VolumeHealth` in only the Controller Plugin, only the Node Plugin, or both if possible.
+
+If `VolumeHealth` is implemented in both the Controller and Node Plugins, it SHALL try to avoid confusion by reporting from different perspectives. 
```

Comment:
Is the intent that the node perspective always overrides that of the controller, or vice-versa?

Or, instead, should CO's be pessimistic and if controller/node are reporting different health conditions then the CO should assume the worst case is the truth?

Asking the CO to merge knowledge of varying health conditions for the same volume seems perilous, but if that's what is expected here then we should probably include a recommendation for COs as to how they should interpret misaligned health conditions (controller and node appear to disagree).

Reply:
Added comments to clarify this.  If controller plugin says the storage system is healthy but node plugin reports there is a mounting problem, they are from different perspective so both are correct.  If in any case they are misaligned, CO should alway pay attention to the worst case.